drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/SettingsPresenter.java

```diff
@@ -97,30 +104,29 @@ public void reset() {
     protected void setRuleSettings(SimulationDescriptor simulationDescriptor) {
         view.getDmnSettings().getStyle().setDisplay(Style.Display.NONE);
         view.getRuleSettings().getStyle().setDisplay(Style.Display.INLINE);
-        view.getDmoSession().setValue(simulationDescriptor.getDmoSession());
-        view.getKieBase().setValue(simulationDescriptor.getKieBase());
-        view.getKieSession().setValue(simulationDescriptor.getKieSession());
-        view.getRuleFlowGroup().setValue(simulationDescriptor.getRuleFlowGroup());
+        view.getKieSession().setValue(Optional.ofNullable(simulationDescriptor.getDmoSession()).orElse(""));
+        view.getRuleFlowGroup().setValue(Optional.ofNullable(simulationDescriptor.getRuleFlowGroup()).orElse(""));
     }
 
     protected void setDMNSettings(SimulationDescriptor simulationDescriptor) {
         view.getRuleSettings().getStyle().setDisplay(Style.Display.NONE);
         view.getDmnSettings().getStyle().setDisplay(Style.Display.INLINE);
-        view.getDmnFilePath().setInnerText(simulationDescriptor.getDmnFilePath());
-        view.getDmnName().setInnerText(simulationDescriptor.getDmnName());
-        view.getDmnNamespace().setInnerText(simulationDescriptor.getDmnNamespace());
+        view.getDmnFilePath().setValue(Optional.ofNullable(simulationDescriptor.getDmnFilePath()).orElse(""));
+        view.getDmnName().setInnerText(Optional.ofNullable(simulationDescriptor.getDmnName()).orElse(""));
+        view.getDmnNamespace().setInnerText(Optional.ofNullable(simulationDescriptor.getDmnNamespace()).orElse(""));
     }
 
     protected void saveRuleSettings() {
-        simulationDescriptor.setDmoSession(view.getDmoSession().getValue());
-        simulationDescriptor.setKieBase(view.getKieBase().getValue());
-        simulationDescriptor.setKieSession(view.getKieSession().getValue());
-        simulationDescriptor.setRuleFlowGroup(view.getRuleFlowGroup().getValue());
+        simulationDescriptor.setDmoSession(getCleanValue(() -> view.getKieSession().getValue()));
+        simulationDescriptor.setRuleFlowGroup(getCleanValue(() -> view.getRuleFlowGroup().getValue()));
     }
 
     protected void saveDMNSettings() {
-        simulationDescriptor.setDmnFilePath(view.getDmnFilePath().getInnerText());
-        simulationDescriptor.setDmnName(view.getDmnName().getInnerText());
-        simulationDescriptor.setDmnNamespace(view.getDmnNamespace().getInnerText());
+        simulationDescriptor.setDmnFilePath(getCleanValue(() -> view.getDmnFilePath().getValue()));
+    }
+
+    private String getCleanValue(Supplier<String> supplier) {
```

Comment:
@danielezonca 
?

Reply:
This method converts empty strings to `null`. Do you prefer something like `getNullIfEmptyString`?

Reply:
@danielezonca 
I'm just not sure if it is worth to add a new method for that, not on the name; anyway keep it as it is, fine for me